registry/data-models/common/models.py

```diff
@@ -0,0 +1,136 @@
+    CATEGORICAL = "categorical"  # Represent a feature that consists of a single category
+    CATEGORICAL_SET = "categorical_set"  # Represent a feature that consists of multiple categories
+    DENSE_VECTOR = "dense_vector"  # Represent a feature in vector format where the majority of the elements are non-zero
+    TERM_VECTOR = "term_vector"  # Represent features that has string terms and numeric value
+    TENSOR = "tensor"  # Represent tensor based features.
+    UNSPECIFIED = "unspecified"  # Placeholder for when no types are specified
+
+
+class Dimension(BaseModel):
+    """
+    Tensor is used to represent feature data. A tensor is a generalization of vectors and matrices to potentially higher dimensions.
+    """
+    type: DimensionType  # Type of the dimension in the tensor. Each dimension can have a different type.
+    shape: Optional[int]  # Size of the dimension in the tensor. If unset, it means the size is unknown and actual size will be determined at runtime.
+
+
+class TensorFeatureFormat(BaseModel):
+    """
+    Defines the format of feature data. Feature data is produced by applying transformation on source, in a Feature.
+    Tensor is used to represent feature data. A tensor is a generalization of vectors and matrices to potentially
+    higher dimensions.
+    """
+    tensorCategory: TensorCategory  # Type of the tensor.
+    valueType: ValueType  # Type of the value column.
+    dimensions: List[Dimension]  # A feature data can have zero or more dimensions (columns that represent keys).
+
+
+class FeatureType(BaseModel):
+    """
+   Information about a featureName. It defines the type, format and default value.
+   Tensor is the next generation representation of the features, so using
+   Tensor type w TensorFeatureFormat would be preferable FeatureType.
+    """
+    type: FeatureValueType  # Defines the high level semantic type of feature.
+    format: Optional[TensorFeatureFormat]  # Defines the format of feature data.
+    defaultValue: Union[bool, int, float, str, bytes]
+
+
+class Clazz(BaseModel):
+    """
+    Reference to a class by fully-qualified name
+    """
+    fullyQualifiedName: str  # A fully-qualified class name including paths.
+
+
+class Function(BaseModel):
+    """
+    Base model for all functions
+    """
+    expression: str  # Expression in str format
+
+
+class MvelExpression(Function):
+    """
+    An expression in MVEL language.
+    """
+    mvel: str  # The MVEL expression
+
+
+class UserDefinedFunction(Function):
+    """
+    User defined function that can be used in feature extraction or derivation.
+    """
+    clazz: Clazz  # Reference to the class that implements the user defined function.
+    parameters: Dict[str, json] = {}  # This field defines the custom parameters of the user defined function
+
+
+class SparkSqlExpression(Function):
+    """
+    An expression in Spark SQL.
+    """
+    sql: str  # Spark SQl expression
```

Comment:
I think my point of reusing `expression` and adding in `type` attribute applies here as well

Reply:
The benefit of doing that is that we can dynamically render common attributes in the UI for newly added classes without much (if at all) code change